js/xt.js

```diff
@@ -0,0 +1,1072 @@
+        //     const market = this.market (this.safeString (ticker, 's'));
+        //     result.push (this.parseTicker (ticker, market));
+        // }
+        return result;
+    }
+
+    parseTicker (response, market) {
+        if (market['spot']) {
+            const percentage = this.safeString (response, 'rate');
+            const marketId = this.safeString (response, 'symbol');
+            market = this.safeMarket (marketId, market, '-');
+            const symbol = market['symbol'];
+            const timestamp = this.safeString (response, 'depthTime');
+            return this.safeTicker ({
+                'symbol': symbol,
+                'timestamp': timestamp,
+                'datetime': this.iso8601 (timestamp),
+                'high': this.safeString (response, 'high'),
+                'low': this.safeString (response, 'low'),
+                'bid': this.safeString (response, 'bid'),
+                'bidVolume': this.safeString (response, 'bidVol'),
+                'ask': this.safeString (response, 'ask'),
+                'askVolume': this.safeString (response, 'askVol'),
+                'vwap': undefined,
+                'open': this.safeString (response, 'open'),
+                'close': this.safeString (response, 'price'),
+                'last': this.safeString (response, 'price'),
+                'previousClose': undefined,
+                'change': undefined,
+                'percentage': percentage,
+                'average': this.safeString (response, 'averagePrice'),
+                'baseVolume': this.safeString (response, 'coinVol'),
+                'quoteVolume': this.safeString (response, 'moneyVol'),
+                'info': response,
+            }, market);
+        } else if (market['swap']) {
+            const ticker = this.safeValue (response, 'result', {});
+            const percentage = this.safeString (ticker, 'r');
+            const marketId = this.safeString (ticker, 's');
+            market = this.safeMarket (marketId, market, '-');
+            const symbol = market['symbol'];
+            const timestamp = this.safeString (ticker, 't');
+            return this.safeTicker ({
+                'symbol': symbol,
+                'timestamp': timestamp,
+                'datetime': this.iso8601 (timestamp),
+                'high': this.safeString (ticker, 'h'),
+                'low': this.safeString (ticker, 'l'),
+                'bid': this.safeString (ticker, 'bp'),
+                'bidVolume': undefined,
+                'ask': this.safeString (ticker, 'ap'),
+                'askVolume': undefined,
+                'vwap': undefined,
+                'open': this.safeString (ticker, 'o'),
+                'close': this.safeString (ticker, 'c'),
+                'last': this.safeString (ticker, 'c'),
+                'previousClose': undefined,
+                'change': String (this.safeNumber (ticker, 'c') - this.safeNumber (ticker, 'o')),
```

Comment:
Three things, `String ()`won't be correctly transpiled to PHP, so we can't use it, and we should be using string-based arithmetic through the `Precise` class, in this case we would do:

```Javascript
Precise.stringSub(this.safeString(), this.safeString())
```

But we shouldn't be doing it 😅 because `safeTicker`will do it for us, so you can't leave it as `undefined`

Reply:
Sorry, but I don't know what you exactly mean by:
> But we shouldn't be doing it because `safeTicker` will do it for us, so you can't leave it as `undefined`

What should I do instead of assigning it as `undefined`?

Reply:
@AlgoQ sorry my bad I made a typo, I meant "you can leave it as undefined"